pkg/controller/endpointslice/reconciler.go

```diff
@@ -299,9 +304,16 @@ func (r *reconciler) reconcileByPortMapping(
 				// otherwise, copy and mark for update
 				epSlice := existingSlice.DeepCopy()
 				epSlice.Endpoints = newEndpoints
+				epSlice.Labels = labels
```

Comment:
consider moving this line to below `setEndpointSliceLabels ` is called

Reply:
this is to prevent other cases were added and people forgot to add the labels.

Reply:
don' t know if I fully understand the comment, but I think that will mutate the current slice ... and IIIRC there was a bug about that, alsodsome of the tests check if the slices are mutated

Reply:
@freehan doing that change several test fails beause of the object mutation

```
=== RUN   TestReconcileEndpointSlicesUpdatePacking
    /home/aojea/go/src/k8s.io/kubernetes/pkg/controller/endpointslice/endpointslice_controller_test.go:872: Cached object was unexpectedly mutated. Original: &EndpointSlice{ObjectMeta:{foo.1  test    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []},Endpoints:[]Endpoint{Endpoint{Addresses:[1.2.3.4],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod0,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.5],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod1,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.6],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod2,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.7],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod3,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.8],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod4,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.9],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod5,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.10],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod6,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.11],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod7,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.12],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod8,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.13],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod9,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.14],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod10,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.15],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod11,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.16],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod12,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.17],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod13,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.18],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod14,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.19],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod15,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.20],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod16,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.21],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod17,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.22],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod18,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.23],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod19,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.24],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod20,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.25],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod21,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.26],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod22,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.27],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod23,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.28],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod24,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.29],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod25,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.30],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod26,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.31],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod27,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.32],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod28,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.33],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod29,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.34],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod30,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.35],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod31,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.36],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod32,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.37],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod33,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.38],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod34,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.39],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod35,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.40],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod36,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.41],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod37,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.42],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod38,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.43],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod39,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.44],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod40,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.45],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod41,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.46],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod42,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.47],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod43,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.48],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod44,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.49],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod45,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.50],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod46,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.51],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod47,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.52],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod48,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.53],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod49,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.54],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod50,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.55],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod51,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.56],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod52,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.57],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod53,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.58],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod54,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.59],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod55,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.60],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod56,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.61],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod57,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.62],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod58,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.63],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod59,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.64],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod60,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.65],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod61,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.66],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod62,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.67],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod63,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.68],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod64,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.69],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod65,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.70],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod66,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.71],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod67,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.72],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod68,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.73],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod69,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.74],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod70,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.75],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod71,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.76],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod72,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.77],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod73,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.78],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod74,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.79],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod75,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.80],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod76,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.81],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod77,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.82],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod78,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.83],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod79,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},},Ports:[]EndpointPort{EndpointPort{Name:*foo,Protocol:*TCP,Port:*80,AppProtocol:nil,},},AddressType:IPv4,}, Mutated: &EndpointSlice{ObjectMeta:{foo.1  test    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[endpointslice.kubernetes.io/managed-by:endpointslice-controller.k8s.io kubernetes.io/service-name:foo service.kubernetes.io/headless:] map[] [] []  []},Endpoints:[]Endpoint{Endpoint{Addresses:[1.2.3.4],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod0,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.5],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod1,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.6],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod2,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.7],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod3,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.8],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod4,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.9],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod5,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.10],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod6,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.11],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod7,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.12],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod8,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.13],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod9,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.14],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod10,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.15],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod11,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.16],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod12,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.17],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod13,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.18],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod14,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.19],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod15,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.20],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod16,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.21],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod17,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.22],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod18,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.23],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod19,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.24],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod20,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.25],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod21,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.26],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod22,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.27],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod23,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.28],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod24,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.29],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod25,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.30],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod26,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.31],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod27,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.32],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod28,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.33],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod29,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.34],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod30,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.35],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod31,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.36],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod32,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.37],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod33,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.38],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod34,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.39],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod35,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.40],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod36,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.41],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod37,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.42],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod38,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.43],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod39,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.44],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod40,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.45],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod41,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.46],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod42,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.47],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod43,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.48],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod44,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.49],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod45,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.50],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod46,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.51],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod47,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.52],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod48,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.53],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod49,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.54],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod50,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.55],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod51,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.56],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod52,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.57],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod53,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.58],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod54,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.59],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod55,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.60],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod56,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.61],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod57,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.62],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod58,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.63],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod59,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.64],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod60,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.65],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod61,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.66],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod62,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.67],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod63,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.68],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod64,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.69],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod65,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.70],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod66,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.71],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod67,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.72],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod68,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.73],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod69,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.74],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod70,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.75],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod71,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.76],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod72,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.77],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod73,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.78],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod74,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.79],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod75,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.80],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod76,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.81],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod77,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.82],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod78,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.83],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod79,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},},Ports:[]EndpointPort{EndpointPort{Name:*foo,Protocol:*TCP,Port:*80,AppProtocol:nil,},},AddressType:IPv4,}
    /home/aojea/go/src/k8s.io/kubernetes/pkg/controller/endpointslice/endpointslice_controller_test.go:872: Cached object was unexpectedly mutated. Original: &EndpointSlice{ObjectMeta:{foo.2  test    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []},Endpoints:[]Endpoint{Endpoint{Addresses:[1.2.3.104],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod100,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.105],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod101,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.106],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod102,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.107],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod103,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.108],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod104,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.109],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod105,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.110],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod106,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.111],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod107,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.112],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod108,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.113],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod109,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.114],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod110,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.115],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod111,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.116],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod112,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.117],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod113,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.118],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod114,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.119],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod115,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.120],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod116,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.121],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod117,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.122],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod118,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.123],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod119,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},},Ports:[]EndpointPort{EndpointPort{Name:*foo,Protocol:*TCP,Port:*80,AppProtocol:nil,},},AddressType:IPv4,}, Mutated: &EndpointSlice{ObjectMeta:{foo.2  test    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[endpointslice.kubernetes.io/managed-by:endpointslice-controller.k8s.io kubernetes.io/service-name:foo service.kubernetes.io/headless:] map[] [] []  []},Endpoints:[]Endpoint{Endpoint{Addresses:[1.2.3.104],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod100,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.105],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod101,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.106],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod102,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.107],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod103,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.108],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod104,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.109],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod105,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.110],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod106,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.111],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod107,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.112],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod108,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.113],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod109,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.114],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod110,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.115],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod111,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.116],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod112,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.117],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod113,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.118],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod114,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.119],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod115,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.120],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod116,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.121],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod117,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.122],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod118,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},Endpoint{Addresses:[1.2.3.123],Conditions:EndpointConditions{Ready:*true,},Hostname:nil,TargetRef:&v1.ObjectReference{Kind:Pod,Namespace:test,Name:pod119,UID:,APIVersion:,ResourceVersion:,FieldPath:,},Topology:map[string]string{kubernetes.io/hostname: node-1,},},},Ports:[]EndpointPort{EndpointPort{Name:*foo,Protocol:*TCP,Port:*80,AppProtocol:nil,},},AddressType:IPv4,}
--- FAIL: TestReconcileEndpointSlicesUpdatePacking (0.01s)
```

Reply:
I think that the case will covered with `labelsChanged`, since it will detect that some of the labels does not match and will update the slice, for creation and deletion we don't have to worry.

Reply:
okay. Fair enough if there are tests around this.